app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -54,26 +71,65 @@ def latest_reports?
       latest_reports.limit(1).present?
     end
 
-    def host_reports_translated_header(shortname, longname)
-      "<th style='width:85px; class='ca'><span class='small' title='' data-original-title='#{longname}'>#{shortname}</span></th>"
+    def failure_hosts_query(format)
+      "report_failures > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
     end
 
-    private
+    def failure_hosts(format)
+      Host.search_for(failure_hosts_query(format)).reorder("").count
+    end
```

Comment:
Every method which returns counts also have `_query` sister method which returns the scoped_search query. The exactly same query is then used to generate the host search link so it is consistent.